tests/state_test.go

```diff
@@ -84,6 +90,9 @@ func createCarmenFactory(t *testing.T) carmenFactory {
 		if err := st.Close(); err != nil {
 			t.Fatalf("cannot close state: %v", err)
 		}
+		if err := os.RemoveAll(dir); err != nil {
```

Comment:
Can I ask you to move this into a `defer` right after the successful creation of the directory? This way, it is going to be removed even if the test fails.

Reply:
ok, I moved it right after the directory was created. It must be in the `Cleanup` callback as the scope outlives this method.